src/client/Microsoft.Identity.Client/AuthenticationResult.cs

```diff
@@ -69,26 +69,42 @@ public partial class AuthenticationResult
             Guid correlationID)
         {
             _authenticationScheme = authenticationScheme ?? throw new ArgumentNullException(nameof(authenticationScheme));
+            string homeAccountId =
+                msalAccessTokenCacheItem?.HomeAccountId ??
+                msalIdTokenCacheItem?.HomeAccountId;
+            string environment = msalAccessTokenCacheItem?.Environment ??
+                msalIdTokenCacheItem?.Environment;
 
-            if (msalAccessTokenCacheItem.HomeAccountId != null)
+            if (homeAccountId != null)
             {
-                string username = msalAccessTokenCacheItem.IsAdfs ? msalIdTokenCacheItem?.IdToken.Upn : msalIdTokenCacheItem?.IdToken?.PreferredUsername;
+                string username = null;
+                if (msalIdTokenCacheItem != null)
+                {
+                    username = msalIdTokenCacheItem.IsAdfs ? 
+                        msalIdTokenCacheItem?.IdToken.Upn : 
+                        msalIdTokenCacheItem?.IdToken?.PreferredUsername;
+                }
+
                 Account = new Account(
-                    msalAccessTokenCacheItem.HomeAccountId,
+                    homeAccountId,
                     username,
-                    msalAccessTokenCacheItem.Environment);
+                    environment);
+            }
+
+            if (msalAccessTokenCacheItem!= null)
```

Comment:
So if the user passed in a scope of "openid", this will simply return the ID token only? Is it the same if the user passes in no scopes at all?

Reply:
MSAL always adds the following 3 scopes to a request: `openid`, `email` and `offline_access`. If the user passes  just`openid` it makes no difference to us, we still send the 3 scopes above. 

So passing 0 scopes or any combination of `openid`, `email`, `offline_access` results in the same behaviour:

- AAD issues an access token, id token, refresh token
- B2C does NOT issue an access token, and MSAL crashes with null reference exceptions.